index.bs

```diff
@@ -2039,6 +2039,27 @@ which meant converting the names to camel-cased version.
 One such example is the <a href="https://w3c.github.io/image-resource/">image resource</a>.
 For this reason, if a key can clearly be expressed as a single word, that is recommended.
 
+<h3 id="debuggability">Features should be developer-friendly</h3>
+
+Any new feature should be developer-friendly.
+While it is hard to quantify friendliness, at least consider the following points.
+
+Error messages must be meaningful.
+When a developer or user encounters an error,
+the message should be specific to that error case,
+and not overly generic.
+
+Ideally, the error message should have enough information
+to guide the developer in pinpointing where the problem is.
```

Comment:
This is somewhat at odds with how we've been developing errors thus far, especially in networking. We tend to use a generic error. Developer consoles ideally give much more specific information, but that's not suitable to directly expose to the web at large.

Reply:
This is for what message an Exception for example should contain. And as the wording says - "ideally". I think It's fine to use a generic error, but I think the message inside that error should be a bit more specific - otherwise developers will search for that message and possibly end up choosing to opt for a suboptimal solution. (or do a lot of trial and error, also not great.)

Reply:
I'm saying that would be problematic as it might expose information about users. We've also had problems with tooling parsing these messages and ending up depending on browser specifics. If anything I'd argue for making things more generic there.

Reply:
I think we might have been talking over each other's heads - @hober pointed this out and the content has been updated in https://github.com/w3ctag/design-principles/pull/264/commits/1a4d5e7411d84d2d42a5c9f2e7a927e36462684c

Reply:
Thanks!